model/grader.py

```diff
@@ -0,0 +1,65 @@
+from .line_segment import LineSegment
+from evaluators.interfaces import LineSegmentEvaluator
+from model.serializer import Serializer
+import json
+from texttable import Texttable
+
+
+class Grader:
+    def __init__(self, evaluator_type: LineSegmentEvaluator, ideal_solution_path: str, predicted_solution_path: str) -> None:
+        self.evalutor = evaluator_type
+        self.ideal_solution_path = ideal_solution_path
+        self.predicted_solution_path = predicted_solution_path
+
+    def _load_json(self, path: str):
+        try:
+            with open(path, 'r') as f:
+                data = json.load(f)
+            return data
+        except Exception as e:
+            raise e
+
+    def calculate_distances(self) -> None:
+        serializer = Serializer()
```

Comment:
you should do this at the init step. `calculate_distances()` should be a pure function that knows nothing about serialization and should not throw errors around serialization.